app/controllers/posts_controller.rb

```diff
@@ -23,6 +25,14 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.author.decrement!(:posts_counter)
+    @post.destroy
+
+    redirect_to root_path, status: :see_other
+  end
+
```

Comment:
- [ ] Kindly make sure to authorize all actions in the posts controller. You can use ``load_and_authorize_resource`` to do so. You can read more about it [here](https://github.com/CanCanCommunity/cancancan#:~:text=can%20be%20read.-,Controller%20helpers,-The%20authorize!).